test/framework/README.md

```diff
@@ -0,0 +1,36 @@
+# CAPI e2e testing framework
+
+This framework aims to define common end-to-end patterns that can be reused across Cluster API providers.
+
+## Usage
+
+Use this framework as you see fit. If there are pieces that don't work for you, do not use them. If there are pieces
+that almost work for you, customize them. If this does not fit your e2e testing please file an issue and we can discuss
+your use case and find a nice solution for everyone.
+
+### Contents
+
+This framework contains
+
+* An interface to a management cluster
+    * An struct that implements the management cluster interface using `kind` as the backend.
+* A series of behavioral tests
+* Clean up tests.
+
+## Requirements
+
+### Code
+
+* Your provider CRDs must implement [`runtime.Object`][runtime.object]. This should already be true if you are using
+[`kubebuilder`][kubebuilder] to generate your controllers and CRDs.
+
+[runtime.object]: https://godoc.org/k8s.io/apimachinery/pkg/runtime#Object
+[kubebuilder]: https://github.com/kubernetes-sigs/kubebuilder
+
+* You must (?) be using ginkgo for your testing framework
```

Comment:
minor - maybe add a link to the _ginkgo_ page?

Reply:
good call, will do